src/app/editor-common/components/expression-model-list/expression-model-list.componen.ts

```diff
@@ -0,0 +1,121 @@
+import {Component, Input, forwardRef} from "@angular/core";
+import {Validators, FormControl, ControlValueAccessor, NG_VALUE_ACCESSOR, FormGroup} from "@angular/forms";
+import {ExpressionModel} from "cwlts/models/d2sb";
+import {ComponentBase} from "../../../components/common/component-base";
+import {GuidService} from "../../../services/guid.service";
+import {CustomValidators} from "../../../validators/custom.validator";
+
+@Component({
+    selector: 'expression-model-list',
+    providers: [
+        { provide: NG_VALUE_ACCESSOR, useExisting: forwardRef(() => ExpressionModelListComponent), multi: true }
+    ],
+    template: `
+        <form *ngIf="form" [formGroup]="form">
+
+            <ol *ngIf="formList.length > 0" class="list-unstyled">
+
+                <li *ngFor="let item of formList"
+                     class="removable-form-control">
+
+                    <ct-expression-input
+                            [context]="context"
+                            [formControl]="form.controls[item.id]">
+                    </ct-expression-input>
+
+                    <div class="remove-icon clickable" (click)="removeExpressionModel(item)">
+                        <i class="fa fa-trash"></i>
+                    </div>
+                </li> 
+            </ol>
+
+            <div *ngIf="formList.length === 0">
+                {{emptyListText}}
+            </div>
+
+            <button type="button" class="btn btn-link add-btn-link no-underline-hover" (click)="addExpressionModel()">
+                <i class="fa fa-plus"></i> {{addButtonText}}
+            </button>
+    </form>
+`
+})
+export class ExpressionModelListComponent extends ComponentBase implements ControlValueAccessor {
```

Comment:
ExpressionModelListComponent is incorrectly implementing ControlValueAccessor. Methods `writeValue`, `registerOnChange` and `registerOnTouch` cannot be private.